ambry-tools/src/main/java/com.github.ambry/tools/admin/BlobValidator.java

```diff
@@ -16,654 +16,445 @@
+    final Operation typeOfOperation;
+
+    /**
+     * The path of the file that contains blob ids. The blob ids should be comma separated with no spaces.
+     * One of this or "blob.ids" must be present.
+     */
+    @Config("blob.ids.file.path")
+    final String blobIdsFilePath;
+
+    /**
+     * Comma separated list of the blob ids to operate on.
+     * One of this or "blob.ids.file.path" must be present.
+     */
+    @Config("blob.ids")
+    final String[] blobIds;
+
+    /**
+     * The hostname of the target server as it appears in the partition layout if applicable.
+     * Applicable to: ValidateBlobOnReplica
+     */
+    @Config("hostname")
+    @Default("localhost")
+    final String hostname;
+
+    /**
+     * The port of the target server in the partition layout (need not be the actual port to connect to) if applicable.
+     * Applicable to: ValidateBlobOnReplica
+     */
+    @Config("port")
+    @Default("6667")
+    final int port;
+
+    /**
+     * The target datacenter if applicable
+     * Applicable to: ValidateBlobOnDatacenter
+     */
+    @Config("datacenter")
+    @Default("")
+    final String datacenter;
+
+    /**
+     * The get option to use to do the get operation (if applicable)
+     * Applicable for: GetBlobProperties,GetUserMetadata,GetBlob
+     */
+    @Config("get.option")
+    @Default("None")
+    final GetOption getOption;
+
+    /**
+     * Replicas to contact per second (for throttling_
+     */
+    @Config("replicas.to.contact.per.sec")
+    @Default("None")
+    final int replicasToContactPerSec;
+
+    /**
+     * Constructs the configs associated with the tool.
+     * @param verifiableProperties the props to use to load the config.
+     */
+    BlobValidatorConfig(VerifiableProperties verifiableProperties) {
+      hardwareLayoutFilePath = verifiableProperties.getString("hardware.layout.file.path");
+      partitionLayoutFilePath = verifiableProperties.getString("partition.layout.file.path");
+      typeOfOperation = Operation.valueOf(verifiableProperties.getString("type.of.operation"));
+      if (verifiableProperties.containsKey("blob.ids.file.path")) {
+        blobIdsFilePath = verifiableProperties.getString("blob.ids.file.path");
+        blobIds = null;
       } else {
-        logger.error("Invalid Type of Operation ");
-        System.exit(1);
-      }
-    } catch (Exception e) {
-      logger.error("Closed with exception ", e);
-    } finally {
-      if (connectionPool != null) {
-        connectionPool.shutdown();
+        blobIdsFilePath = null;
+        blobIds = verifiableProperties.getString("blob.ids").split(",");
       }
+      hostname = verifiableProperties.getString("hostname", "localhost");
+      port = verifiableProperties.getIntInRange("port", 6667, 1, 65535);
+      datacenter = verifiableProperties.getString("datacenter", "");
+      getOption = GetOption.valueOf(verifiableProperties.getString("get.option", "None"));
+      replicasToContactPerSec =
```

Comment:
have you tried running the tool with all different inputs (diff operations, diff get options, blobIdfilePath and so on). If you don't have time, let me know. I will try to run it to ensure everything works fine.